src/jsoncharutils.h

```diff
@@ -306,6 +306,1014 @@ inline size_t codepoint_to_utf8(uint32_t cp, uint8_t *c) {
+    {0x8da471a9de737e24L, 1821}, {0xb10d8e1456105dadL, 1824},
+    {0xdd50f1996b947518L, 1827}, {0x8a5296ffe33cc92fL, 1831},
+    {0xace73cbfdc0bfb7bL, 1834}, {0xd8210befd30efa5aL, 1837},
+    {0x8714a775e3e95c78L, 1841}, {0xa8d9d1535ce3b396L, 1844},
+    {0xd31045a8341ca07cL, 1847}, {0x83ea2b892091e44dL, 1851},
+    {0xa4e4b66b68b65d60L, 1854}, {0xce1de40642e3f4b9L, 1857},
+    {0x80d2ae83e9ce78f3L, 1861}, {0xa1075a24e4421730L, 1864},
+    {0xc94930ae1d529cfcL, 1867}, {0xfb9b7cd9a4a7443cL, 1870},
+    {0x9d412e0806e88aa5L, 1874}, {0xc491798a08a2ad4eL, 1877},
+    {0xf5b5d7ec8acb58a2L, 1880}, {0x9991a6f3d6bf1765L, 1884},
+    {0xbff610b0cc6edd3fL, 1887}, {0xeff394dcff8a948eL, 1890},
+    {0x95f83d0a1fb69cd9L, 1894}, {0xbb764c4ca7a4440fL, 1897},
+    {0xea53df5fd18d5513L, 1900}, {0x92746b9be2f8552cL, 1904},
+    {0xb7118682dbb66a77L, 1907}, {0xe4d5e82392a40515L, 1910},
+    {0x8f05b1163ba6832dL, 1914}, {0xb2c71d5bca9023f8L, 1917},
+    {0xdf78e4b2bd342cf6L, 1920}, {0x8bab8eefb6409c1aL, 1924},
+    {0xae9672aba3d0c320L, 1927}, {0xda3c0f568cc4f3e8L, 1930},
+    {0x8865899617fb1871L, 1934}, {0xaa7eebfb9df9de8dL, 1937},
+    {0xd51ea6fa85785631L, 1940}, {0x8533285c936b35deL, 1944},
+    {0xa67ff273b8460356L, 1947}, {0xd01fef10a657842cL, 1950},
+    {0x8213f56a67f6b29bL, 1954}, {0xa298f2c501f45f42L, 1957},
+    {0xcb3f2f7642717713L, 1960}, {0xfe0efb53d30dd4d7L, 1963},
+    {0x9ec95d1463e8a506L, 1967}, {0xc67bb4597ce2ce48L, 1970},
+    {0xf81aa16fdc1b81daL, 1973}, {0x9b10a4e5e9913128L, 1977},
+    {0xc1d4ce1f63f57d72L, 1980}, {0xf24a01a73cf2dccfL, 1983},
+    {0x976e41088617ca01L, 1987}, {0xbd49d14aa79dbc82L, 1990},
+    {0xec9c459d51852ba2L, 1993}, {0x93e1ab8252f33b45L, 1997},
+    {0xb8da1662e7b00a17L, 2000}, {0xe7109bfba19c0c9dL, 2003},
+    {0x906a617d450187e2L, 2007}, {0xb484f9dc9641e9daL, 2010},
+    {0xe1a63853bbd26451L, 2013}, {0x8d07e33455637eb2L, 2017},
+    {0xb049dc016abc5e5fL, 2020}, {0xdc5c5301c56b75f7L, 2023},
+    {0x89b9b3e11b6329baL, 2027}, {0xac2820d9623bf429L, 2030},
+    {0xd732290fbacaf133L, 2033}, {0x867f59a9d4bed6c0L, 2037},
+    {0xa81f301449ee8c70L, 2040}, {0xd226fc195c6a2f8cL, 2043},
+    {0x83585d8fd9c25db7L, 2047}, {0xa42e74f3d032f525L, 2050},
+    {0xcd3a1230c43fb26fL, 2053}, {0x80444b5e7aa7cf85L, 2057},
+    {0xa0555e361951c366L, 2060}, {0xc86ab5c39fa63440L, 2063},
+    {0xfa856334878fc150L, 2066}, {0x9c935e00d4b9d8d2L, 2070},
+    {0xc3b8358109e84f07L, 2073}, {0xf4a642e14c6262c8L, 2076},
+    {0x98e7e9cccfbd7dbdL, 2080}, {0xbf21e44003acdd2cL, 2083},
+    {0xeeea5d5004981478L, 2086}, {0x95527a5202df0ccbL, 2090},
+    {0xbaa718e68396cffdL, 2093}, {0xe950df20247c83fdL, 2096},
+    {0x91d28b7416cdd27eL, 2100}, {0xb6472e511c81471dL, 2103},
+    {0xe3d8f9e563a198e5L, 2106}, {0x8e679c2f5e44ff8fL, 2110}};
+
+// A complement from power_of_ten_components
+// complete to a 128-bit mantissa.
+const uint64_t mantissa_128[] = {0x419ea3bd35385e2d,
+                                 0x52064cac828675b9,
```

Comment:
it looks like we can use `// clang-format on` and `// clang-format off` to disable formatting for a section, in case you'd be interested in doing that